src/corpora/auto.py

```diff
@@ -58,103 +84,98 @@ def get_auto_dataset(
-        for k in dataset
-    }
-    # Create Parent Path of Cache Files
-    (paths["preprocessed"] / dataset_id / "preprocessing" / "tokenization").mkdir(parents=True, exist_ok=True)
+        def tokenize(examples: Dict[str, List[str]]) -> BatchEncoding:
+            return tokenizer(examples["text"])
 
-    tokenized_dataset = dataset.map(
-        tokenize,
-        batched=True,
-        num_proc=preprocessing_num_proc,
-        remove_columns=next(iter(dataset.values())).column_names,
-        cache_file_names=post_tokenization_cache_files,
-        load_from_cache_file=True,
-    )
+        tokenized_dataset = dataset.map(
+            tokenize,
+            batched=True,
+            num_proc=preprocessing_num_proc,
+            remove_columns=next(iter(dataset.values())).column_names,
+            cache_file_names=post_tokenization_cache_files,
+            load_from_cache_file=True,
+        )
 
     # Finally, actually run chunking (collapse multiple sequences into a giant document to read `seq_len` chunks from)
-    def group(examples: Dict[str, Iterable[List[int]]]) -> Dict[str, List[List[int]]]:
-        # Concatenate all the Texts
-        concatenated: Dict[str, List[int]] = {k: sum(examples[k], []) for k in examples.keys()}
-        total_length = len(concatenated[list(examples.keys())[0]])
-
-        # Drop the "very last" bit of the dataset that doesn't fit into block size...
-        total_length = ((total_length - seq_len + stride) // stride) * stride
-
-        # Split by Chunks of Maximum Length
-        result = {k: [t[i : i + seq_len] for i in range(0, total_length, stride)] for k, t in concatenated.items()}
-        result["labels"] = deepcopy(result["input_ids"])
-
-        # Mask out losses in overlapping regions. If training data, string will be equal to seq_len
-        for i, labels in enumerate(result["labels"]):
-            if i == 0:
-                continue
-            for j in range(len(labels) - stride):
-                labels[j] = -100
-            result["labels"][i] = labels
-        return result
+    group = _GrouperFun(seq_len, stride)
 
     # From HF.Examples :: Note that with `batched=True`, this map processes 1,000 texts together, so group_texts throws
     # away a remainder for each of those groups of 1,000 texts. You can adjust that batch_size here but a higher
     # value might be slower to preprocess.
     #   - Sidd Note (3/11): We're dropping a max of 8M / 9B tokens... we're probably fine!
-    overwatch.info(f"Auto-Batching Dataset via Multiprocessing with `{preprocessing_num_proc}` threads...")
-
-    # Create Post-Chunking Cache Paths
-    post_chunking_cache_files = {
-        k: str(paths["preprocessed"] / dataset_id / "preprocessing" / "chunking" / f"{k}-stride={stride}-chunked.hf")
-        for k in dataset
-    }
-    # Create Parent Path of Cache Files
-    (paths["preprocessed"] / dataset_id / "preprocessing" / "chunking").mkdir(parents=True, exist_ok=True)
+    if streaming:
```

Comment:
Ok at this point I'm wondering if we should drop HF Datasets?

Reply:
yeah i certainly think we should not try to rely on its streaming support, and possibly everything. What mosaic does is pretty gross but maybe this is what it needs to look like? https://github.com/stanford-crfm/composer/blob/pubmed/composer/datasets/c4.py#L127